aten/src/ATen/core/ivalue.cpp

```diff
@@ -470,6 +470,57 @@ std::shared_ptr<ClassType> ivalue::Object::type() const {
   return type_.type_->expect<ClassType>();
 }
 
+IValue IValue::copy() const {
+  IValue copy;
+  switch(tag) {
+    case IValue::Tag::Tensor:
+      copy = IValue(toTensor());
+      break;
+    case IValue::Tag::Tuple: {
+      std::vector<IValue> copied_tuple;
+      for (const auto& e : toTuple()->elements()) {
+        copied_tuple.push_back(e.copy());
```

Comment:
A shallow copy shouldn't be copying the subreferences right?

Reply:
I see, will change